### PR TITLE
Trace Analytics support for custom sources

### LINKED ISSUE #2052: [FEATURE] Trace overview page and enable CCS for tracing data
**Is your feature request related to a problem?**
create a simple filtering view without analytics to view traces and service maps

**What solution would you like?**
Phase 1

* The trace analytics landing page will be redesigned to allow seamless rendering of traces, spans and service maps. As part of this redesign, we will be adding the following capabilities:
    * Attributes/Metadata Search: Capability to search trace metadata and list/filter attributes.
    * Filters from Traces: Functionality to apply filters directly from services and service map.
    * Paginate Results: Limit the number of pages to 5+ by default, with the option to load more on request, to improve page loading times.
    * Column Selector: Selector for attributes and metadata columns.
    * Overlays: No default service map calculations. Combo-box to add overlays for
        * Duration
        * Error
        * Request rate + top-level filter.
    * Search Source and Selected Fields: Ability to search the source and specific fields.
    * Trace Details View: Clicking shows detailed view of traces including a detailed trace page with Gantt chart and service map.
* Trace analytics plugin will support wildcards in index names for traces. dataprepper pipelines will generate dynamic index naming based on incoming metadata in trace spans and index names should be dynamically generated based on this metadata.
* Trace analytics will support CCS allowing customers to search for traces across their regional cluster from their central proxy cluster.
Screenshot 2024-08-09 at 3 09 04 PM
Screenshot 2024-08-09 at 3 09 12 PM

Phase 2

* The UX will have an advanced mode wherein pre-aggregations performed by dataprepper pipelines will be rendered on the dashboard. This will allow customers to offload the compute intensive task of aggregations to the dataprepper pipelines while still having the ability to see the aggregated results in the Trace analytics view. User will be able to switch from default/simple experience to this advanced experience in Trace Analytics view.

### PR DESCRIPTION
**Pull request by ps48**
### Description
Experimental feature for Trace Analytics support of custom sources

### Issues Resolved
#2052 

* Update mode picker with custom trace source 
Screenshot 2024-09-03 at 5 33 59 PM

* Flyout to manage/update custom trace source
Screenshot 2024-09-03 at 5 34 07 PM

* Add service filtering widgets to traces view
Screenshot 2024-09-03 at 5 34 24 PM

* Lazy load red metrics 
Screenshot 2024-09-03 at 5 34 52 PM

* Hide/Show attribute columns
Screenshot 2024-09-03 at 5 36 01 PM

* Added custom trace indices in user settings 
Screenshot 2024-09-03 at 5 37 43 PM

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/trace_analytics/components/common/plots/service_map.tsx [hunk 1]
```diff
@@ -93,6 +97,35 @@ export function ServiceMap({
...
+  const [selectableValue, setSelectableValue] = useState<Array<EuiSuperSelectOption<any>>>([]);
+
+  const onChangeSelectable = (value: React.SetStateAction<Array<EuiSuperSelectOption<any>>>) => {
+    // if the change is changing for the first time then callback servicemap with metrics
+    if (selectableValue.length === 0 && value.length !== 0) {
+      if (includeMetricsCallback) {
+        includeMetricsCallback();
+      }
+    }
+    // console.log('value[0].id', value);
```
Comment (TackAdam) [inline public/components/trace_analytics/components/common/plots/service_map.tsx:109]: Missed deleting console log comment.

**Comment by TackAdam** [comment]
UI looks correct / functional under the new navigation.
Screenshot 2024-09-03 at 6 41 35 PM

**Comment by TackAdam** [comment]
Screenshot 2024-09-03 at 6 35 58 PM

Under the new navigation when services and traces are broken apart, do we need to change this drop down for services?

**Comment by TackAdam** [comment]
Screenshot 2024-09-03 at 6 44 08 PM

Spacing seems a little off between the drop downs. Not sure if we have a reference for multiple bars.
Not blocking.

**Review by TackAdam — APPROVED** [review]
Great work!

**Review by RyanL1997 — APPROVED** [review]

**Comment by ps48** [comment]
@TackAdam @RyanL1997  Thanks for the review. 
> Spacing seems a little off between the drop downs. Not sure if we have a reference for multiple bars.
* Updated spacing for flex items
Screenshot 2024-09-03 at 9 42 05 PM


> Under the new navigation when services and traces are broken apart, do we need to change this drop down for services?
* Changed it to custom source

**Pull request merged**